hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -350,22 +351,45 @@ object LoweredTableReader {
           body: IR => IR): TableStage = {
           val partOrigIndex = sortedPartData.map(_.getInt(6))
 
-          val partitioner = new RVDPartitioner(pkType,
+          val partitioner = new RVDPartitioner(keyType,
             sortedPartData.map { partData =>
-              Interval(selectPK(partData.getAs[Row](0)), selectPK(partData.getAs[Row](1)), includesStart = true, includesEnd = true)
+              Interval(selectPK(partData.getAs[Row](1)), selectPK(partData.getAs[Row](2)), includesStart = true, includesEnd = true)
             }, key.length)
 
           val pkPartitioned = TableStage(globals, partitioner,
-            Literal(TArray(contextType), partOrigIndex.map(i => contexts(i))),
+            ToStream(Literal(TArray(contextType), partOrigIndex.map(i => contexts(i)))),
             body)
 
-          throw new LowererUnsupportedOperation("pksorted VCF import requires repartition")
+          pkPartitioned
+            .strictify()
+            .mapPartition(None) { part =>
+              flatMapIR(StreamGroupByKey(part, pkType.fieldNames)) { inner =>
+                ToStream(sortIR(inner) { case (l, r) => ApplyComparisonOp(LT(l.typ), l, r) })
+              }
+            }
```

Comment:
This probably works, but it does break the invariant that a TableStage is sorted by its key. I think it would be better to move this to a method on TableStage, `localSort` or something, that takes a stage keyed by `pkType` and returns a stage keyed by `keyType`, so the invariant is never violated.

Reply:
it's easy to move the logic around so we preserve the invariant.